drivers/ethernet/eth_lan865x.c

```diff
@@ -0,0 +1,584 @@
+	if (ret < 0) {
+		return -ENODEV;
+	}
+
+	ctx->silicon_rev = val & LAN865X_REV_MASK;
+	if (ctx->silicon_rev != 1 && ctx->silicon_rev != 2) {
+		return -ENODEV;
+	}
+
+	ctx->chip_id = (val >> 4) & 0xFFFF;
+	if (ctx->chip_id != LAN8650_DEVID && ctx->chip_id != LAN8651_DEVID) {
+		return -ENODEV;
+	}
+
+	return ret;
+}
+
+/* Implementation of pseudo code from AN1760 */
+static uint8_t lan865x_read_indirect_reg(const struct device *dev, uint8_t addr,
+					 uint8_t mask)
+{
+	struct lan865x_data *ctx = dev->data;
+	uint32_t val;
+
+	oa_tc6_reg_write(ctx->tc6, 0x000400D8, addr);
+	oa_tc6_reg_write(ctx->tc6, 0x000400DA, 0x02);
+
+	oa_tc6_reg_read(ctx->tc6, 0x000400D9, &val);
+
+	return (uint8_t) val & mask;
+}
+
+static int lan865x_init_chip(const struct device *dev, uint8_t silicon_rev)
+{
+	struct lan865x_data *ctx = dev->data;
+	uint8_t value1, value2;
+	int8_t offset1 = 0, offset2 = 0, ret;
+	uint16_t value3, value4, value5, value6, value7;
+	uint16_t cfgparam1, cfgparam2, cfgparam3, cfgparam4, cfgparam5;
+	uint32_t val;
+
+	ret = lan865x_read_indirect_reg(dev, 0x05, 0x40);
+	if (ret == 0) {
+		LOG_ERR("LAN865x error! Please contact microchip support for replacement.");
+		return -EIO;
+	}
+
+	value1 = lan865x_read_indirect_reg(dev, 0x04, 0x1F);
+	if ((value1 & 0x10) != 0) { /* Convert uint8_t to int8_t */
+		offset1 = value1 | 0xE0;
+		if (offset1 < -5) {
+			LOG_ERR("LAN865x internal error!");
+			return -EIO;
+		}
+	} else {
+		offset1 = value1;
+	}
+
+	value2 = lan865x_read_indirect_reg(dev, 0x08, 0x1F);
+	if ((value2 & 0x10) != 0) { /* Convert uint8_t to int8_t */
+		offset2 = value2 | 0xE0;
+	} else {
+		offset2 = value2;
+	}
+
+	oa_tc6_reg_read(ctx->tc6, 0x00040084, &val);
+	value3 = (uint16_t)val;
+
+	oa_tc6_reg_read(ctx->tc6, 0x0004008A, &val);
+	value4 = (uint16_t)val;
+
+	oa_tc6_reg_read(ctx->tc6, 0x000400AD, &val);
+	value5 = (uint16_t)val;
+
+	oa_tc6_reg_read(ctx->tc6, 0x000400AE, &val);
+	value6 = (uint8_t)val;
+
+	oa_tc6_reg_read(ctx->tc6, 0x000400AF, &val);
+	value7 = (uint8_t)val;
+
+	cfgparam1 = (value3 & 0xF) | (((9 + offset1) << 10) | ((14 + offset1) << 4));
+	cfgparam2 = (value4 & 0x3FF) | ((40 + offset2) << 10);
+	cfgparam3 = (value5 & 0xC0C0) | (((5 + offset1) << 8) | (9 + offset1));
+	cfgparam4 = (value6 & 0xC0C0) | (((9 + offset1) << 8) | (14 + offset1));
+	cfgparam5 = (value7 & 0xC0C0) | (((17 + offset1) << 8) | (22 + offset1));
+
+	oa_tc6_reg_write(ctx->tc6, 0x00040084, (uint32_t) cfgparam1);
+	oa_tc6_reg_write(ctx->tc6, 0x0004008A, (uint32_t) cfgparam2);
+	oa_tc6_reg_write(ctx->tc6, 0x000400AD, (uint32_t) cfgparam3);
+	oa_tc6_reg_write(ctx->tc6, 0x000400AE, (uint32_t) cfgparam4);
+	oa_tc6_reg_write(ctx->tc6, 0x000400AF, (uint32_t) cfgparam5);
+
+	return 0;
+}
+/* Implementation of pseudo code from AN1760 - END */
+
+static int lan865x_config_plca(const struct device *dev, uint8_t node_id,
+			       uint8_t node_cnt, uint8_t burst_cnt, uint8_t burst_timer)
+{
+	struct lan865x_data *ctx = dev->data;
+	uint32_t regVal;
```

Comment:
regVal name looks suspicious, is it to follow specific name from standard? Otherwise it should not be camelCase.

Reply:
Ok, I will fix it